### DISCUSSION TITLE
where to get mbtiles files from to use offline with mapsui

### DISCUSSION THREAD

**Discussion by gigfy2** [Mapsui/Mapsui#2363] [Q&A] [answered]
I've been jumping from site to site trying to figure out how to get or make high definition satellite mbtiles files that can be used offline with mapsui.  I've had zero luck so far.  Any tips?

**Comment by gigfy2** [comment, marked as answer]
QGIS is the answer.  Download the desktop app.  Go to XYZ Tiles in the browser.  Add a connection.  I wanted to use bing hybrid.  That URL is http://ecn.t2.tiles.virtualearth.net/tiles/h{q}.jpeg?g=1

For some reason, they don't have that listed, so you have to add it manually.  They have many others already with the URLs.

Then on the Processing toolbox go to Raster tools

double click on Generate XYZ tiles (MBTiles)

The Extent is the area you want.  I do the draw on canvas so I can get exactly the area I want.  Minimum zoom is 0, Max depends on how much area you are going for, and how much zoom you need.  Then run it and the resulting mbtiles file(s) work with mapsui.  If you have multiple files, like a world file and a more detailed area, you can create multiple map layers in mapsui and it will stack them.

I never got a response here when I was trying to figure out how to make mbtiles files that mapsui would use without issue, so hopefully this will help people.  Also QGIS is free.

**Reply by pauldendulk** [reply]
Thank you very much! This is great feedback.
